[CI] BinaryFieldMapperTests testSyntheticSource failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/s7erfwvnsaztw/tests/:server:test/org.elasticsearch.index.mapper.BinaryFieldMapperTests/testSyntheticSource


**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.index.mapper.BinaryFieldMapperTests.testSyntheticSource" -Dtests.seed=76CB47079EE58437 -Dtests.locale=ar-QA -Dtests.timezone=America/Montreal -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
[Failure dashboard for `org.elasticsearch.index.mapper.BinaryFieldMapperTests#testSyntheticSource`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testSyntheticSource'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.index.mapper.BinaryFieldMapperTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: round trip {"field":["Lb4wahfHIexuatbriBOD4YSrE8yUACSIQdEwwLvf9nug94Vo5PwdWOOuSCMOFOzVo5nCvVgsR/GgLofebDwB2l65raY+GsU7O4c+5reXWnQb5sXknoP5YFZ9m4VpiguobsyWL4/hXpmfEjp4EnmqSDhs+43sns8ZcbdxMJtsS6JZlYWbfbErkN4=","mCMfDh34p7GPqRV65cJoIcR3wjS6n2GPt3g1jRIJ7VHb0hcIbp/nahm4RgYZ+412QKdNR1BgHBSsmEnRzZDAYkLuvEGI9nmY9r33Cf4i1IlgJt8HKFNIDhJyq8Cyn5QCTEC/mKUNTTrpF57xbjPjnAtq39YylSzeaYgvZvhhrMUksbt0RxaWf2Ba/37VVcH+aiU9/ZKkCtghS3hCSJ6hNDym2N6Ww7MA7sICuxe6BDOeJGH6CNf8/7FVK37l5kgzJVcg1v5Ly6Itqgzr6HI7Y98jgXN0Uw42LgfzCZ7UmgqTU4UrUR67","4t4e7jMKXQ8EGdaE1kaCgo49dsoe68r4ttCvf6KTX+25wWRA8GOHcMqq5HXzYgyJe9oVjei+9yDmWB6trXulmiGrlf09FDD+FQwHHTSqlKV6wGuCSQoGED6lCekSG17N370x29P8hHRUasCfwN721kPqeOsqQR01JFWliftJVIwB4HWFTVsbyAzjzjadAJIYFS+CfGlLnhP9LK0q7ebyGtj+y3AhBFzyQuVV2eRtcdR5rsCrNvBZ+sTtlxrlhR39Aq3JddDONd/C5qSsGqsex3a4oc1Ilrgrc/opOg==","6DUTuDA2K4ouRICvL5PQ3OG7UGIWKTZgYMUQOaWjehX3V3HLg8ubNmR/IxPWMuoDR/i2Ex3lSCC9pzKdUJI="]} expected:<[e2 de 1e ee 33 a 5d f 4 19 d6 84 d6 46 82 82 8e 3d 76 ca 1e eb ca f8 b6 d0 af 7f a2 93 5f ed b9 c1 64 40 f0 63 87 70 ca aa e4 75 f3 62 c 89 7b da 15 8d e8 be f7 20 e6 58 1e ad ad 7b a5 9a 21 ab 95 fd 3d 14 30 fe 15 c 7 1d 34 aa 94 a5 7a c0 6b 82 49 a 6 10 3e a5 9 e9 12 1b 5e cd df bd 31 db d3 fc 84 74 54 6a c0 9f c0 de f6 d6 43 ea 78 eb 2a 41 1d 35 24 55 a5 89 fb 49 54 8c 1 e0 75 85 4d 5b 1b c8 c e3 ce 36 9d 0 92 18 15 2f 82 7c 69 4b 9e 13 fd 2c ad 2a ed e6 f2 1a d8 fe cb 70 21 4 5c f2 42 e5 55 d9 e4 6d 71 d4 79 ae c0 ab 36 f0 59 fa c4 ed 97 1a e5 85 1d fd 2 ad c9 75 d0 ce 35 df c2 e6 a4 ac 1a ab 1e c7 76 b8 a1 cd 48 96 b8 2b 73 fa 29 3a]> but was:<[98 23 1f e 1d f8 a7 b1 8f a9 15 7a e5 c2 68 21 c4 77 c2 34 ba 9f 61 8f b7 78 35 8d 12 9 ed 51 db d2 17 8 6e 9f e7 6a 19 b8 46 6 19 fb 8d 76 40 a7 4d 47 50 60 1c 14 ac 98 49 d1 cd 90 c0 62 42 ee bc 41 88 f6 79 98 f6 bd f7 9 fe 22 d4 89 60 26 df 7 28 53 48 e 12 72 ab c0 b2 9f 94 2 4c 40 bf 98 a5 d 4d 3a e9 17 9e f1 6e 33 e3 9c b 6a df d6 32 95 2c de 69 88 2f 66 f8 61 ac c5 24 b1 bb 74 47 16 96 7f 60 5a ff 7e d5 55 c1 fe 6a 25 3d fd 92 a4 a d8 21 4b 78 42 48 9e a1 34 3c a6 d8 de 96 c3 b3 0 ee c2 2 bb 17 ba 4 33 9e 24 61 fa 8 d7 fc ff b1 55 2b 7e e5 e6 48 33 25 57 20 d6 fe 4b cb a2 2d aa c eb e8 72 3b 63 df 23 81 73 74 53 e 36 2e 7 f3 9 9e d4 9a a 93 53 85 2b 51 1e bb]>

  at __randomizedtesting.SeedInfo.seed([76CB47079EE58437:C02D4A47156977F6]:0)
  at org.junit.Assert.fail(Assert.java:89)
  at org.junit.Assert.failNotEquals(Assert.java:835)
  at org.junit.Assert.assertEquals(Assert.java:120)
  at org.apache.lucene.tests.util.LuceneTestCase.assertStoredFieldEquals(LuceneTestCase.java:2504)
  at org.apache.lucene.tests.util.LuceneTestCase.assertStoredFieldsEquals(LuceneTestCase.java:2494)
  at org.apache.lucene.tests.util.LuceneTestCase.assertReaderEquals(LuceneTestCase.java:2082)
  at org.elasticsearch.index.mapper.MapperServiceTestCase.validateRoundTripReader(MapperServiceTestCase.java:749)
  at org.elasticsearch.index.mapper.MapperServiceTestCase.roundTripSyntheticSource(MapperServiceTestCase.java:722)
  at org.elasticsearch.index.mapper.MapperServiceTestCase.syntheticSource(MapperServiceTestCase.java:698)
  at org.elasticsearch.index.mapper.MapperTestCase.assertSyntheticSource(MapperTestCase.java:1132)
  at org.elasticsearch.index.mapper.MapperTestCase.testSyntheticSource(MapperTestCase.java:1115)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/BinaryFieldMapperTests.java
@@ -255,7 +255,7 @@ public class BinaryFieldMapperTests extends MapperTestCase {
                 b.field("type", "binary").field("doc_values", "true");
 
                 if (rarely()) {
-                    b.field("store", true);
+                    b.field("store", false);
                 }
             }
 

